swap/src/protocol/alice/swap.rs

```diff
@@ -75,13 +75,25 @@ async fn run_until_internal(
 
     let new_state = match state {
         AliceState::Started { state3 } => {
-            timeout(env_config.bob_time_to_act, tx_lock_status.wait_until_seen())
-                .await
-                .context("Failed to find lock Bitcoin tx")??;
-
-            tx_lock_status.wait_until_final().await?;
-
-            AliceState::BtcLocked { state3 }
+            match timeout(
+                env_config.bitcoin_lock_confirmed_timeout,
+                tx_lock_status.wait_until_final(),
+            )
```

Comment:
Should we just make this timeout part of `wait_until_final`? We could introduce a dedicated error type that represents the timeout so the user can still decide on what to do in case we don't want this for every `wait_until_final`.

Reply:
Going to hold off for now because there is no type safety preventing  `.wait_until_final()` from being called on other transactions. Won't really make sense for any other transaction to timeout with `bitcoin_lock_confirmed timeout`.